packages/react/src/components/numberInput/NumberInput.test.tsx

```diff
@@ -72,4 +72,30 @@ describe('<NumberInput /> spec', () => {
     });
     expect(onChange.mock.calls.length).toBe(1);
   });
+
+  it('should be able to reset a controlled NumberInput to an empty value', async () => {
+    const TestComponent = () => {
+      const [value, setValue] = useState<'' | number>(10);
+      return (
+        <div>
+          <button
+            type="button"
+            onClick={() => {
+              setValue('');
+            }}
+          >
+            Reset number input
+          </button>
+          <NumberInput step={10} value={value} {...numberInputProps} />
```

Comment:
An observation:
I noticed that this would require the onChange handler to be functional.
I tested it in a story, and I needed to add: 
```js
 onChange={(e) => {
    const eventValue = e.target.value;
    const newValue = eventValue === '' ? eventValue : parseInt(eventValue, 10);
    setValue(newValue);
 }}
```